eth2deposit/credentials.py

```diff
@@ -48,55 +48,54 @@ def verify_keystore(self, keystore_filefolder: str, password: str) -> bool:
-    '''
-    Given a DepositMessage, it signs its root and returns a Deposit
-    '''
-    assert bls.PrivToPub(sk) == deposit_data.pubkey
-    domain = compute_domain()
-    signing_root = compute_signing_root(deposit_data, domain)
-    signed_deposit_data = Deposit(
-        **deposit_data.as_dict(),
-        signature=bls.Sign(sk, signing_root)
-    )
-    return signed_deposit_data
-
-
-def export_deposit_data_json(*, credentials: List[ValidatorCredentials], folder: str) -> str:
-    deposit_data: List[Dict[bytes, bytes]] = []
-    for credential in credentials:
-        deposit_datum = DepositMessage(
-            pubkey=credential.signing_pk,
-            withdrawal_credentials=SHA256(credential.withdrawal_pk),
-            amount=credential.amount,
+    def sign_deposit_data(self, deposit_message: DepositMessage) -> Deposit:
+        '''
+        Given a DepositMessage, it signs its root and returns a DepositData
+        '''
+        assert bls.PrivToPub(self.signing_sk) == deposit_message.pubkey
+        domain = compute_domain()
+        signing_root = compute_signing_root(deposit_message, domain)
+        signed_deposit_data = Deposit(
+            **deposit_message.as_dict(),
+            signature=bls.Sign(self.signing_sk, signing_root)
         )
-        signed_deposit_datum = sign_deposit_data(deposit_datum, credential.signing_sk)
-        datum_dict = signed_deposit_datum.as_dict()
-        datum_dict.update({'deposit_data_root': deposit_datum.hash_tree_root})
-        datum_dict.update({'signed_deposit_data_root': signed_deposit_datum.hash_tree_root})
-        deposit_data.append(datum_dict)
-
-    filefolder = os.path.join(folder, 'deposit_data-%i.json' % time.time())
-    with open(filefolder, 'w') as f:
-        json.dump(deposit_data, f, default=lambda x: x.hex())
-    return filefolder
-
-
-def verify_keystores(*, credentials: List[ValidatorCredentials],
-                     keystore_filefolders: List[str], password: str) -> bool:
-    return all(credential.verify_keystore(keystore_filefolder=filefolder, password=password)
-               for credential, filefolder in zip(credentials, keystore_filefolders))
+        return signed_deposit_data
+
+
+class CredentialList:
+    def __init__(self, credentials: List[Credential]):
+        self.credentials = credentials
+
+    @classmethod
+    def from_mnemonic(cls, *, mnemonic: str, num_keys: int, amounts: List[int], start_index: int=0) -> 'CredentialList':
+        assert len(amounts) == num_keys
+        key_indices = range(start_index, start_index + num_keys)
+        return cls([Credential(mnemonic=mnemonic, index=index, amount=amounts[index])
+                    for index in key_indices])
+
+    def export_keystores(self, password: str, folder: str) -> List[str]:
+        return [credential.save_signing_keystore(password=password, folder=folder) for credential in self.credentials]
+
+    def export_deposit_data_json(self, folder: str) -> str:
```

Comment:
We can use`py-ssz` API to encode/decode SSZ object <> json string format. See the example [here](https://github.com/ethereum/py-ssz/blob/d3edd4d759076ea3d12ffb8ea0ce4d90921d1ee5/tests/tools/test_parsing_and_dumping.py#L11-L14).  [Remerkleable also has JSON converter now](https://github.com/protolambda/remerkleable/blob/a8e6824a0d40f8c127b6bb09f1fb403946d7a610/remerkleable/test_impl.py#L341-L349)

The SSZ libraries above use the same format as the SSZ test suite format. The bytes data would be encoded as '0x' prefix hex string. (py-ssz decoder accepts both '0x' prefix or no '0x' hex string).

Maybe it's better to use the same format with py-ssz API here?

Reply:
I'd like to implement this change with a simultaneous update to the `ethereum/eth2.0-deposit` repo where `@chainsafe/ssz`is used to check the ssz calculations (instead of just using the root passed in via the `deposit_data.json`).

Reply:
ahh I see, the `0x` prefix will be added on launchpad. Got it 👌